src/components/Editor/index.js

```diff
@@ -201,24 +202,38 @@ class Editor extends PureComponent {
     // `this.props` to be the current props. This lifecycle method is
     // responsible for updating the editor annotations.
     const { selectedLocation, selectedSource } = this.props;
-
+    if (!selectedLocation || !selectedLocation.location) {
+      return;
+    }
     // If the location is different and a new line is requested,
     // update the pending jump line. Note that if jumping to a line in
     // a source where the text hasn't been loaded yet, we will set the
     // line here but not jump until rendering the actual source.
-    if (prevProps.selectedLocation !== selectedLocation) {
-      if (selectedLocation && selectedLocation.line != undefined) {
+
+    if (
+      !prevProps.selectedLocation ||
+      prevProps.selectedLocation.location !== selectedLocation.location
```

Comment:
will this work? 

* do we need to use: https://lodash.com/docs/#isEqual or should identity comparisons be enough
* naive question: why is it `prevProps.selectedLocation.location` and not `prevProps.selectedLocation`? Are we messing something up?